pkg/server/endpoints/authorized_entryfetcher_test.go

```diff
@@ -205,6 +229,56 @@ func TestBuildCacheSavesSkippedEvents(t *testing.T) {
 	assert.ElementsMatch(t, expectedMetrics, metrics.AllMetrics(), "should emit no metrics")
 }
 
+func TestRunUpdateCacheTaskDoesFullUpdate(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	log, _ := test.NewNullLogger()
+	log.SetLevel(logrus.DebugLevel)
+	clk := clock.NewMock(t)
+	ds := fakedatastore.New(t)
+	metrics := fakemetrics.New()
+
+	ef, err := NewAuthorizedEntryFetcherEvents(ctx, AuthorizedEntryFetcherEventsConfig{
+		log:                     log,
+		metrics:                 metrics,
+		clk:                     clk,
+		ds:                      ds,
+		cacheReloadInterval:     2 * time.Second,
+		fullCacheReloadInterval: 3 * time.Second,
+		pruneEventsOlderThan:    defaultPruneEventsOlderThan,
+		eventTimeout:            defaultEventTimeout,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, ef)
+
+	ef.mu.RLock()
+	initialCache := ef.cache
+	ef.mu.RUnlock()
+
+	// Start Update Task
+	updateCacheTaskErr := make(chan error)
+	go func() {
+		updateCacheTaskErr <- ef.RunUpdateCacheTask(ctx)
+	}()
+	clk.WaitForTickerMulti(time.Second, 2, "waiting to create tickers")
+
+	// First iteration, cache should not be rebuilt
+	clk.Add(2 * time.Second)
+	ef.mu.RLock()
+	require.Equal(t, initialCache, ef.cache)
+	ef.mu.RUnlock()
+
+	// Second iteration, cache should be rebuilt
+	clk.Add(2 * time.Second)
+	ef.mu.RLock()
+	require.NotEqual(t, initialCache, ef.cache)
```

Comment:
This is maybe enough but I wonder if it's possible to test the case where the full rebuild actually helps? For example creating a registration entry and deleting the event (or however these unit tests manage to make skipped events) and then waiting for the full reload to read that registration entry.